backend/redisCache.js

```diff
@@ -0,0 +1,87 @@
+const redis = require("redis");
+const { promisify } = require("util");
+
+class RedisCache {
+  constructor(ttl = 3600) {
+    // Create a Redis client with built-in Promise support
+    this.client = redis.createClient({
+      legacyMode: true, // This enables legacy mode which includes the callback API
```

Comment:
i'm not familiar with Redis - why do we want the callback API when it seems like we're using promises for everything?

Reply:
I've actually switched from this approach to another in a later PR